### PR TITLE
Automation for CEPH-83573868

### PR DESCRIPTION
**Pull request by sumabai**
# Description
CEPH-83573868 - Root squash option in MultiFS
https://docs.ceph.com/en/latest/cephfs/client-auth/#root-squash

    Test Steps:
    1. Create 2 CepFS and subvolume in each FS
    2. Create client user with root_squash cap for / and regular permissions for /volumes on both CephFS and verify
    3. Perform Kernel and Fuse mount of CephFS
    4. Validate root_squash is working on / and write permissions work on /volumes 
    5. cleanup
    
Suites:
suites/pacific/cephfs/tier-2_cephfs_test-clients.yaml
suites/reef/cephfs/tier-2_cephfs_test-clients.yaml

Logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-KLJRBL

Latest logs with review comments addressed : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-K6CUNG/verify_root_squash_cap_works_in_multiFS_0.log

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/clients/verify_root_squash_in_caps.py [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+                    cmd_str = f"{cmd} {mnt_path}/{cmds_to_run[cmd]}"
+                    if cmd in ["mkdir", "touch"]:
+                        cmd_str += f"_{suffix}"
+                    try:
+                        out, err = client.exec_command(sudo=True, cmd=cmd_str)
+                        assert (
+                            False
+                        ), "Write ops succeed in path that has root_squash enabled, its Unexpected"
+                    except CommandFailed:
+                        log.info(out)
```
Comment (neha-gangadhar) [inline tests/cephfs/clients/verify_root_squash_in_caps.py:145]: As a practice, log.error() is good to use in exception blocks.

File: tests/cephfs/clients/verify_root_squash_in_caps.py [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+        log.info("Checking if the root_squash path / has read permissions")
+        for fs_name in fs_list:
+            for mnt_path in mount_path[fs_name].values():
+                cmd_str = f" cat {mnt_path}/{filename}"
+                try:
+                    out, err = client.exec_command(sudo=True, cmd=cmd_str)
+                    log.info(out)
+                except CommandFailed:
+                    log.info(out)
```
Comment (neha-gangadhar) [inline tests/cephfs/clients/verify_root_squash_in_caps.py:155]: As a practice, log.error() is good to use in exception blocks.

File: tests/cephfs/clients/verify_root_squash_in_caps.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        clients = ceph_cluster.get_ceph_objects("client")
+        client = clients[0]
+        log.info("checking Pre-requisites")
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+
+        log.info(
+            "Create Test configuration - Additional CephFS and one subvolume in each CephFS"
+        )
+        client.exec_command(sudo=True, cmd="ceph fs volume create cephfs_1")
```
Comment (AmarnatReddy) [inline tests/cephfs/clients/verify_root_squash_in_caps.py:37]: Can you use create_fs() function from https://github.com/red-hat-storage/cephci/blob/d4f64519f8faa90475abebf2b4904b1e52dfbc3e/tests/cephfs/cephfs_utilsV1.py#L1133

File: tests/cephfs/clients/verify_root_squash_in_caps.py [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+        fs_list = ["cephfs", "cephfs_1"]
```
Comment (AmarnatReddy) [inline tests/cephfs/clients/verify_root_squash_in_caps.py:38]: there might be chances of cephfs not present in the cluster.Can we add validation for it 
```
fs_details = fs_util.get_fs_info(client)
        if not fs_details:
            fs_util.create_fs(client, "cephfs")
```

File: tests/cephfs/clients/verify_root_squash_in_caps.py [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+            fs_util.kernel_mount(
+                [client],
+                mount_path[fs_name]["kernel"],
+                ",".join(mon_node_ips),
+                new_client_hostname="verify_root_squash",
+            )
+            mon_ips = ",".join(mon_node_ips)
+            mount_cmd = f"mount -t ceph {mon_ips}:/ {mount_path[fs_name]['kernel']} "
+            mount_cmd += f"-o name=verify_root_squash,secretfile=/etc/ceph/verify_root_squash.secret,fs={fs_name}"
+            client.exec_command(sudo=True, cmd=mount_cmd)
```
Comment (AmarnatReddy) [inline tests/cephfs/clients/verify_root_squash_in_caps.py:112]: From line no: 109 t0 112 be replaced with below kernel mount call
```
fs_util.kernel_mount(
                [client],
                mount_path[fs_name]["kernel"],
                ",".join(mon_node_ips),
                new_client_hostname="verify_root_squash",
                extra_params=f",fs={fs_name}",
            )
```

File: tests/cephfs/clients/verify_root_squash_in_caps.py (hunk 3 above)
Comment (sumabai) [reply tests/cephfs/clients/verify_root_squash_in_caps.py]: addressed

File: tests/cephfs/clients/verify_root_squash_in_caps.py (hunk 5 above)
Comment (sumabai) [reply tests/cephfs/clients/verify_root_squash_in_caps.py]: added

**Pull request closed without merging**